Mathijs testrun continued

MathijsMul opened the pull request.

Commit:
Use screenshot of CICD pipeline

Commit:
Use screenshot of CICD pipeline

MathijsMul commented:
https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/6fa0a33a84a91ce06250b3280119a97a102a5202/environment_setup/CICD-dev-template.yml#L64

Staat het antwoord voor de TODO hier niet iets te letterlijk?

Commit:
Show all TODOs in screenshot CI pipeline

Commit:
Update screenshot

Commit:
Add Databricks workspace screenshot

larsjanssen-valcon commented:
> https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/6fa0a33a84a91ce06250b3280119a97a102a5202/environment_setup/CICD-dev-template.yml#L64
> 
> Staat het antwoord voor de TODO hier niet iets te letterlijk?

Ja klopt, het zijn ook wel specifieke manieren om variabelen te referencen in workflow scripts. Weet jij misschien een alternatief om de deelnemer iets meer uit te dagen?

larsjanssen-valcon commented:
> > https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/6fa0a33a84a91ce06250b3280119a97a102a5202/environment_setup/CICD-dev-template.yml#L64
> > 
> > Staat het antwoord voor de TODO hier niet iets te letterlijk?
> 
> Ja klopt, het zijn ook wel specifieke manieren om variabelen te referencen in workflow scripts. Weet jij misschien een alternatief om de deelnemer iets meer uit te dagen?

We kunnen deze [link](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow) meegeven en aangeven dat de variables als $DBX_TOKEN en $DBX_ORG_URL gereferenced moeten kunnen worden.

Commit:
Extend instructions

Commit:
Test linked service before creating

MathijsMul commented:
[**@larsjanssen-valcon**](https://github.com/larsjanssen-valcon) commented on [19 apr. 2022 15:47 CEST](https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/pull/5#issuecomment-1102678517 "2022-04-19T13:47:18Z - Replied by Github Reply Comments"):
> > https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/6fa0a33a84a91ce06250b3280119a97a102a5202/environment_setup/CICD-dev-template.yml#L64
> > 
> > Staat het antwoord voor de TODO hier niet iets te letterlijk?
> 
> Ja klopt, het zijn ook wel specifieke manieren om variabelen te referencen in workflow scripts. Weet jij misschien een alternatief om de deelnemer iets meer uit te dagen?

MathijsMul commented:
[**@larsjanssen-valcon**](https://github.com/larsjanssen-valcon) commented on [19 apr. 2022 15:47 CEST](https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/pull/5#issuecomment-1102678517 "2022-04-19T13:47:18Z - Replied by Github Reply Comments"):
> > https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/6fa0a33a84a91ce06250b3280119a97a102a5202/environment_setup/CICD-dev-template.yml#L64
> > 
> > Staat het antwoord voor de TODO hier niet iets te letterlijk?
> 
> Ja klopt, het zijn ook wel specifieke manieren om variabelen te referencen in workflow scripts. Weet jij misschien een alternatief om de deelnemer iets meer uit te dagen?

Ik zou die regel 64 gewoon weglaten. En dan mss die link idd. Dan hebben ze nog iets uit te vogelen.

MathijsMul commented:
https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/248cea49011d52a98fdbc9c2d403154d8383ed6d/README.md?plain=1#L281

vs

https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/248cea49011d52a98fdbc9c2d403154d8383ed6d/README.md?plain=1#L296

We gebruiken hier het Production train script en het Development validate script. Willen we die niet allebei uit Production of allebei uit Development halen?

Commit:
Create cluster if none exist yet

MathijsMul commented:
Ik krijg een SyntaxError als het train notebook runt: 

```
SyntaxError: f-string: empty expression not allowed
  File "<[command-960008263229533]()>", line 4
    experiment_path = f"{workspace_dir}/Experiments/{}".format(model_name)
                      ^
SyntaxError: f-string: empty expression not allowed
```

Commit:
Fix typo

Commit:
Fix f-string

Commit:
Make training work

MathijsMul commented on scripts/train.py:
```diff
@@ -134,15 +135,14 @@
-    except:
+        print("Experiment path does not yet exist; will create it now.")
+
+        # For now set the experiment name equal to the given model_name
```
Ik heb dit zo aangepast omdat het voor mij anders niet werkte. Het zou netter zijn om hier alsnog wel een "Experiments" directory te hebben; dat is nu namelijk weg.

MathijsMul commented:
Train pipeline gaat bij mij nog steeds niet goed. Krijg deze error: 

```
RestException: INVALID_PARAMETER_VALUE: Expected id '0' to be an MLflow experiment or Databricks notebook, found 'DIRECTORY'. NOTE: The default MLflow experiment has been migrated and no longer has experiment id '0'. For more information, see 'https://docs.databricks.com/applications/mlflow/experiments.html#experiment-migration'.
```

Commit:
Register model during training

MathijsMul commented on README.md:
```diff
@@ -284,7 +284,7 @@ Now that we configured the Data Factory, we need to create the train pipeline th
       1. Configure the parameter
          1. Name: train_type
-         2. Value: adf_pipeline
+         2. Value: register_model
```
In het train notebook is dit de waarde die 'train_type' moet hebben om een model te registreren. Dat willen we wel, dus moeten we hier niet dit hebben ipv 'adf_pipeline'?

spanmandewit-valcon replied:
Ja, dit was hiervoor adf_pipeline. Dus volgens mij heb je het goed aangepast.

MathijsMul commented:
https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/1647965ba1d13974ebcbd27fd33da50f4f49296c/scripts/validate.py#L10-L32

Hier in het validate script gaan we ervan uit dat er al een model is in "Production". Maar als je het train notebook voor de eerste keer hebt gerund en verder niks hebt gedaan, dan staat er nog geen model in productie en krijg je dus errors bij het validate notebook. 

Dus we moeten:
* ofwel het eerst getrainde model in het train notebook automatisch doorzetten naar "Production";
* of het validate notebook niet laten afhangen van de aanwezigheid van een "Production" model

MathijsMul commented:
[**@MathijsMul**](https://github.com/MathijsMul) commented on [20 apr. 2022 15:50 CEST](https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/pull/5#issuecomment-1103956712 "2022-04-20T13:50:02Z - Replied by Github Reply Comments"):
> https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/1647965ba1d13974ebcbd27fd33da50f4f49296c/scripts/validate.py#L10-L32
> 
> Hier in het validate script gaan we ervan uit dat er al een model is in "Production". Maar als je het train notebook voor de eerste keer hebt gerund en verder niks hebt gedaan, dan staat er nog geen model in productie en krijg je dus errors bij het validate notebook.
> 
> Dus we moeten:
> 
> *   ofwel het eerst getrainde model in het train notebook automatisch doorzetten naar "Production";
> *   of het validate notebook niet laten afhangen van de aanwezigheid van een "Production" model

In deze screenshot zie je ook al eerdere modellen staan, en idd eentje in "Production":

https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/1647965ba1d13974ebcbd27fd33da50f4f49296c/README.md?plain=1#L322

Maar voor de deelnemers aan de workshop staat er na het uitvoeren van het train script alleen nog maar 1 model met stage "None".

MathijsMul commented:
https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/1647965ba1d13974ebcbd27fd33da50f4f49296c/scripts/score.py#L22

Ook hier veronderstellen we dat er al een model met stage "Production" is.

MathijsMul commented:
https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/1647965ba1d13974ebcbd27fd33da50f4f49296c/scripts/score.py#L135

Het scoren werkt bij mij nog niet. Ik krijg TypeErrors: 
```
TypeError: Not supported type for data.<class 'xgboost.core.DMatrix'>
```

MathijsMul commented:
https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/1647965ba1d13974ebcbd27fd33da50f4f49296c/README.md?plain=1#L353-L354

Waar moeten deze links naar verwijzen? Als ik erop klik blijf ik op dezelfde pagina.

lpajanssen-sbp committed:
Change CICD-dev-template.yml instructions

larsjanssen-valcon committed:
Minor instruction change CICD-dev-template.yml

Commit:
Update README.md

Fixed consistency in using Production train and validate scripts

Commit:
Update train.py

Fixed issue when running training script from ADF

larsjanssen-valcon committed:
Change wrap-up text README.md

larsjanssen-valcon commented:
> https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/1647965ba1d13974ebcbd27fd33da50f4f49296c/README.md?plain=1#L353-L354
> 
> Waar moeten deze links naar verwijzen? Als ik erop klik blijf ik op dezelfde pagina.

Ik heb de links verwijderd en de volgorde van de wrap-up geswitcht.

lpajanssen-sbp committed:
Change validate.py to process case when 0 models in prod

larsjanssen-valcon commented:
> https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/blob/1647965ba1d13974ebcbd27fd33da50f4f49296c/scripts/validate.py#L10-L32
> 
> Hier in het validate script gaan we ervan uit dat er al een model is in "Production". Maar als je het train notebook voor de eerste keer hebt gerund en verder niks hebt gedaan, dan staat er nog geen model in productie en krijg je dus errors bij het validate notebook.
> 
> Dus we moeten:
> 
> * ofwel het eerst getrainde model in het train notebook automatisch doorzetten naar "Production";
> * of het validate notebook niet laten afhangen van de aanwezigheid van een "Production" model

Aangepast naar eerste bullet point suggestie, zie [commit](https://github.com/larsjanssen-valcon/MLOps-GitHubActions-Valcon/pull/5/commits/163d595765c5faf550aeb1b4df7deffb673b5842).

Commit:
updated screenshot with newest variable

Commit:
Updated readme,

Deleted parameters for validate script (this script does not use a widget anymore)
